openedx/core/djangoapps/user_api/views.py

```diff
@@ -279,6 +283,16 @@ def get(self, request):
                     required=self._is_field_required(field_name)
                 )
 
+        # Add a data sharing consent dialog if it's requested
+        if active_provider_requests_data_sharing(request):
+            enforced = active_provider_enforces_data_sharing(request, EnterpriseCustomer.AT_LOGIN)
+            provider_name = get_enterprise_customer_for_request(request).name
+
+            # data sharing consent checkbox is required only when provider enforces it and
+            # does not want the user account created otherwise. that is why we are setting
+            # "required" html attribute only when providers enforces data sharing consent.
+            self._add_data_sharing_consent_field(form_desc, provider_name, required=enforced)
```

Comment:
For the moment, I'm testing without that entire structure, but it seems like a good thing to incorporate.